src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c

```diff
@@ -1,27 +1,128 @@
 /*
  * File: z_eff_ss_hitmark.c
  * Overlay: ovl_Effect_Ss_Hitmark
- * Description:
+ * Description: Hit Marks
  */
 
 #include "z_eff_ss_hitmark.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rTexIndex regs[0]
+#define rType regs[1]
+#define rPrimColorR regs[2]
+#define rPrimColorG regs[3]
+#define rPrimColorB regs[4]
+#define rEnvColorR regs[5]
+#define rEnvColorG regs[6]
+#define rEnvColorB regs[7]
+#define rScale regs[8]
 
 #define PARAMS ((EffectSsHitmarkInitParams*)initParamsx)
 
-s32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsHitmark_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsHitmark_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
+static Color_RGB8 sColors[] = {
+    { 255, 255, 255 }, { 255, 255, 0 }, { 255, 255, 255 }, { 255, 0, 0 },   { 255, 200, 100 }, { 200, 150, 0 },
+    { 150, 100, 0 },   { 100, 50, 0 },  { 255, 255, 255 }, { 255, 0, 0 },   { 255, 255, 0 },   { 255, 0, 0 },
+    { 255, 255, 255 }, { 0, 255, 200 }, { 255, 255, 255 }, { 150, 0, 255 },
+};
+
+static TexturePtr sTextures[] = {
+    gEffHitmark1Tex,  gEffHitmark2Tex,  gEffHitmark3Tex,  gEffHitmark4Tex,  gEffHitmark5Tex,  gEffHitmark6Tex,
+    gEffHitmark7Tex,  gEffHitmark8Tex,  gEffHitmark9Tex,  gEffHitmark10Tex, gEffHitmark11Tex, gEffHitmark12Tex,
+    gEffHitmark13Tex, gEffHitmark14Tex, gEffHitmark15Tex, gEffHitmark16Tex, gEffHitmark17Tex, gEffHitmark18Tex,
+    gEffHitmark19Tex, gEffHitmark20Tex, gEffHitmark21Tex, gEffHitmark22Tex, gEffHitmark23Tex, gEffHitmark24Tex,
+    gEffHitmark1Tex,  gEffHitmark2Tex,  gEffHitmark3Tex,  gEffHitmark4Tex,  gEffHitmark5Tex,  gEffHitmark6Tex,
+    gEffHitmark7Tex,  gEffHitmark8Tex,
+};
+
 const EffectSsInit Effect_Ss_Hitmark_InitVars = {
     EFFECT_SS_HITMARK,
     EffectSsHitmark_Init,
 };
 
-#endif
+u32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsHitmarkInitParams* initParams = PARAMS;
+    s32 colorIndex;
+
+    this->pos = initParams->pos;
+    this->gfx = gEffHitmarkDL;
+
+    if (initParams->type == EFFECT_HITMARK_DUST) {
+        this->life = 16;
+    } else {
+        this->life = 8;
+    }
+
+    this->draw = EffectSsHitmark_Draw;
+    this->update = EffectSsHitmark_Update;
+
+    colorIndex = initParams->type * 4;
+    this->rTexIndex = 0;
+    this->rType = initParams->type;
+    this->rPrimColorR = sColors[colorIndex].r;
+    this->rPrimColorG = sColors[colorIndex].g;
+    this->rPrimColorB = sColors[colorIndex].b;
+    this->rEnvColorR = sColors[colorIndex + 1].r;
+    this->rEnvColorG = sColors[colorIndex + 1].g;
+    this->rEnvColorB = sColors[colorIndex + 1].b;
```

Comment:
```suggestion
    this->rPrimColorR = sColors[colorIndex][0].r;
    this->rPrimColorG = sColors[colorIndex][0].g;
    this->rPrimColorB = sColors[colorIndex][0].b;
    this->rEnvColorR = sColors[colorIndex][1].r;
    this->rEnvColorG = sColors[colorIndex][1].g;
    this->rEnvColorB = sColors[colorIndex][1].b;
```
Could this also match as a 2D array?
`static Color_RGB8 sColors[8][2] = {